File: A1_q2.c
/*
 * C program to find the factorial of a number provided by user
 */
#include <stdio.h>
 
  int main()
{
	//intialise the value
    int i, num, factorial = 1;
 
    printf("\nEnter an integer number \n");
    scanf ("\n%d", &num );
    
    //check if the number is negative,positive or zero.  
    if (num<0)
    {
    	printf("\nFactorial does not exist for negative integer.Enter the positive integer.\n");
	}
	else if(num==0) 
	{
		printf("\nFactorial of 0 is 1.");
	}
	else
    {
    	
	//loop iterate till num
       for (i = 1; i <= num; i++)
        {
        factorial = factorial * i;
        }
      
    printf ("\nFactorial of %d is = %d \n", num, factorial);
    }
}

File: A1_Q2.py
# find the factorial of a number provided by the user.

# taking the input from the user.
num = int(input("Enter the number : "))
factorial = 1

# check if the number is negative, positive or zero
if num < 0:
   print("Factorial does not exist for negative numbers. Enter the positive number.")
elif num == 0:
   print("The factorial of 0 is 1")
else:
#loop iterate till num
     for i in range(1,num+1 ):
       factorial = factorial*i
print("The factorial of",num,"is",factorial)

